cachi2/core/package_managers/npm.py

```diff
@@ -585,30 +625,68 @@ def _update_package_json_files(
     return package_json_projectfiles
 
 
+def _merge_npm_sbom_properties(components: list[NpmComponentInfo]) -> list[NpmComponentInfo]:
+    """Deduplicate the NpmComponentInfo objects and merge their dev and bundled properties."""
+    merged_components: dict[str, NpmComponentInfo] = {}
+    for component in components:
+        purl = component["purl"]
+        if purl not in merged_components:
+            merged_components[purl] = component
+            continue
+
+        # The dev and bundled properties for a package are True only if True for all
+        # components with the same purl
+        merged_component = merged_components[purl]
+        merged_component["bundled"] = component.get("bundled", False) and merged_component.get(
+            "bundled", False
+        )
+        merged_component["dev"] = component.get("dev", False) and merged_component.get("dev", False)
```

Comment:
nitpick: no need to `.get()` the required keys of a TypedDict
```suggestion
        merged_component["bundled"] = component["bundled"] and merged_component["bundled"]
        merged_component["dev"] = component["dev"] and merged_component["dev"]
```